src/toil/jobStores/azureJobStore.py

```diff
@@ -266,15 +266,23 @@ def writeStatsAndLogging(self, statsAndLoggingString):
                                                      encrypted=str(encrypted)))
         self.statsFileIDs.insert_entity(entity={'RowKey': jobStoreFileID})
 
-    def readStatsAndLogging(self, statsAndLoggingCallbackFn):
+    def readStatsAndLogging(self, statsAndLoggingCallbackFn, readAll=False):
         numStatsFiles = 0
         for entity in self.statsFileIDs.query_entities():
             jobStoreFileID = entity.RowKey
-            with self._downloadStream(jobStoreFileID, self.statsFiles) as fd:
-                statsAndLoggingCallbackFn(fd)
-            self.statsFiles.delete_blob(blob_name=jobStoreFileID)
-            self.statsFileIDs.delete_entity(row_key=jobStoreFileID)
-            numStatsFiles += 1
+            hasNotBeenRead = False if len(jobStoreFileID) > len(str(uuid.uuid4())) else True
```

Comment:
`False if c else True` is equivalent to `not c`

Don't like the negation in the variable name. Just makes it harder to reason about.